opensearch-project/security
# securityadmin: Replace TransportClient by RestHighLevelClient

### Description
[Describe what this change achieves]
* Category = Maintenance
* Why these changes are required?
Check description in: https://github.com/opensearch-project/security/issues/1578

### Issues Resolved
https://github.com/opensearch-project/security/issues/1578

### Testing
Testing done: 
- unit testing
- manual tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
